src/main/java/com/bhoruka/bloodbank/dao/CampDao.java

```diff
@@ -20,4 +29,57 @@
     public CampDao(@NonNull final CampRepository campRepository) {
         this.campRepository = campRepository;
     }
+
+    /**
+     * Checks if a camp exists in the Camp table.
+     * @param campId id of the camp
+     * @return true if camp exists, false otherwise
+     */
+    public Boolean campExists(@NonNull final String campId) {
+        return campRepository.existsById(CampId.builder()
+                .id(campId)
+                .build());
+    }
+
+    /**
+     * Fetches a camp from the Camp table.
+     * @param campId id of the camp
+     * @return Optional containing the CampModel or empty if camp does not exist in the db
+     */
+    public Optional<CampModel> getCamp(@NonNull final String campId) {
+        Optional<Camp> campOptional = campRepository.findById(CampId.builder()
+                .id(campId)
+                .build());
+
+        if (!campOptional.isPresent()) {
+            log.info("Unable to find camp with id : {}", campId);
```

Comment:
Isn't this an error instead of info?

Reply:
We can't assume that it is an error as it might just be that an incorrect camp id had been passed